lib/modules/platform/bitbucket-server/utils.ts

```diff
@@ -148,3 +149,28 @@ export function getExtraCloneOpts(opts: HostRule): GitOptions {
   }
   return {};
 }
+
+export function splitEscapedSpaces(str: string): string[] {
+  const parts = str.split(' ');
+  const result: string[] = [];
+  let last: string | undefined;
+
+  for (const part of parts) {
+    if (last?.endsWith(String.raw`\\`)) {
+      result[result.length - 1] = last.slice(0, -2) + ' ' + part;
+    } else {
+      result.push(part);
+    }
+    last = result[result.length - 1];
+  }
+
+  return result;
+}
+
+export function parseModifier(value: string): number | null {
+  const match = regEx('^random(?:\\((\\d+)\\))?$').exec(value);
+  if (!match) {
+    return null;
+  }
+  return parseInt(match[1] ?? '1', 10);
```

Comment:
```suggestion
  return parseInt(match[1] ?? '1');
```

`10` is the default radix since ecma 5